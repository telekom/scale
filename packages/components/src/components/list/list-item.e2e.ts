import { newE2EPage } from '@stencil/core/testing';

describe('scale-list-item', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent('<scale-list-item>default</scale-list-item>');
    const element = await page.find('scale-list-item');
    expect(element).toHaveClass('hydrated');
  });
});
