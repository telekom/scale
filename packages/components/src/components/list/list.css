/**
 * @license
 * Scale https://github.com/telekom/scale
 *
 * Copyright (c) 2021 Egor Kirpichev and contributors, Deutsche Telekom AG
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

:host {
  --spacing-left: 0;
  --spacing-left-nested: var(--telekom-spacing-unit-x-2);
}

.list {
  padding-left: var(--spacing-left);
}

.list--nested {
  margin-top: 0;
  margin-bottom: 0;
  padding-left: var(--spacing-left-nested);
}
