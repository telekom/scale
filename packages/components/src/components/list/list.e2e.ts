import { newE2EPage } from '@stencil/core/testing';

describe('scale-list', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent('<scale-list>default</scale-list>');
    const element = await page.find('scale-list');
    expect(element).toHaveClass('hydrated');
  });
});
