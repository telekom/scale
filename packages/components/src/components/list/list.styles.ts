import { JssStyle } from 'jss';

export const styles: JssStyle = {
  list: {
    paddingLeft: '1.5rem',
  },
};
