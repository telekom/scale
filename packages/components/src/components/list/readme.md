# scale-list

<!-- Auto Generated Below -->


## Properties

| Property  | Attribute | Description                           | Type      | Default |
| --------- | --------- | ------------------------------------- | --------- | ------- |
| `ordered` | `ordered` | (optional) Make the list ordered (ol) | `boolean` | `false` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
