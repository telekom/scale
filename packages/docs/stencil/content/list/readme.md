# scale-list

<!-- Auto Generated Below -->


## Properties

| Property  | Attribute | Description             | Type                                     | Default       |
| --------- | --------- | ----------------------- | ---------------------------------------- | ------------- |
| `styles`  | --        |                         | `StyleSheet<string \| number \| symbol>` | `undefined`   |
| `variant` | `variant` | (optional) List variant | `string`                                 | `'unordered'` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
