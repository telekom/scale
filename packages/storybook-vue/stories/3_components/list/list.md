<div style="display: inline-flex; align-items: center; justify-content: space-between; width: 100%;">
    <h1>Text List</h1>
    <img src="assets/aa.png" alt="Accessible AA" />
</div>

A list arranges related pieces of information in a vertical group. This makes it easier for users to scan the content for relevant information.

## General

You can choose between an ordered, unordered, or definition list.

![Image Name](assets/3_components/text-list/text-list_general.png)

### When to use

Use a list to:

- avoid large bulks of information in paragraphs,
- highlight content that would otherwise be lost in the body text,
- communicate complicated content or processes step by step, or
- provide the definition of terms.

### When not to use

Emotionally charged content is not suitable to present as a list.

## Basic types

### Ordered list

You can convey a specific hierarchy by using an ordered list. The list items follow a series. Indent items to display subordinate information.

![Image Name](assets/3_components/text-list/ordered_list.png)

### Unordered list

The items of an unordered list have equal status. The list will not convey a specific order. Nevertheless, you should organize the items in a logical way that users can easily understand (e.g., alphabetically or chronologically). Even if there is no specific order, you can display different levels of hierarchy by indenting.

![Image Name](assets/3_components/text-list/unordered_list.png)

### Definition list

> Available soon

A definition list consists of terms and their definition.

![Image Name](assets/3_components/text-list/definition_list.png)

## Related components

<a href="?path=/usage/components-table--standard">Table</a>
