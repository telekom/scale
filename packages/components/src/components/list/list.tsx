import { Component, h, Prop, Host } from '@stencil/core';
import { CssClassMap } from '../../utils/utils';
import classNames from 'classnames';
import { styles } from './list.styles';
import { CssInJs } from '../../utils/css-in-js';
import { StyleSheet } from 'jss';
import Base from '../../utils/base-interface';
@Component({
  tag: 'scale-list',
  shadow: true,
})
export class List implements Base {
  /** (optional) List variant */
  @Prop() variant?: string = 'unordered';
  @Prop() styles?: StyleSheet;
  /** decorator Jss stylesheet */
  @CssInJs('List', styles) stylesheet: StyleSheet;
  componentWillLoad() {}
  componentWillUpdate() {}
  componentDidUnload() {}
  render() {
    let Tag;

    switch (this.variant) {
      case 'ordered':
        Tag = 'ol';
        break;
      case 'definition':
        Tag = 'dl';
        break;
      default:
        Tag = 'ul';
        break;
    }

    return (
      <Host>
        <style>{this.stylesheet.toString()}</style>
        <Tag class={this.getCssClassMap()}>
          <slot />
        </Tag>
      </Host>
    );
  }
  getCssClassMap(): CssClassMap {
    const { classes } = this.stylesheet;
    return classNames(
      classes.list,
      this.variant && classes[`list--variant-${this.variant}`]
    );
  }
}
