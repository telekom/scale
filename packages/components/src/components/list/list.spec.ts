import { newSpecPage } from '@stencil/core/testing';
import { List } from './list';
import { styles } from './list.styles';
import jss from 'jss';

describe('List', () => {
  let element;
  let stylesheet;
  beforeEach(async () => {
    element = new List();
    stylesheet = element.stylesheet = jss.createStyleSheet(styles as any);
  });

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [List],
      html: `<scale-list>default</scale-list>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should handle css classes', () => {
    element.variant = 'ordered';
    stylesheet.addRule('list--variant-ordered', {});
    expect(element.getCssClassMap()).toContain(
      stylesheet.classes['list--variant-ordered']
    );
  });
});
