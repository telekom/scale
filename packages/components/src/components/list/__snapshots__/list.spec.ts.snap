// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should match snapshot 1`] = `
<scale-list>
  <mock:shadow-root>
    <style>
      .list-0-0-2 {
  padding-left: 1.5rem;
}
    </style>
    <ul class="list-0-0-2">
      <slot></slot>
    </ul>
  </mock:shadow-root>
  default
</scale-list>
`;
